plugins/imu/include/yarp/dev/IMUDriver.h

```diff
@@ -55,24 +60,80 @@ class yarp::dev::GazeboYarpIMUDriver:
+    bool calibrate(int channelIndex, double v) override;
 
     //PRECISELY TIMED
-    virtual yarp::os::Stamp getLastInputStamp();
+    yarp::os::Stamp getLastInputStamp() override;
+
+    /* IThreeAxisGyroscopes methods */
+
+    size_t getNrOfThreeAxisGyroscopes() const override;
+
+    yarp::dev::MAS_status getThreeAxisGyroscopeStatus(size_t sens_index) const override;
+
+    bool getThreeAxisGyroscopeName(size_t sens_index, std::string &name) const override;
+
+    bool getThreeAxisGyroscopeFrameName(size_t sens_index, std::string &frameName) const override;
+
+    bool getThreeAxisGyroscopeMeasure(size_t sens_index, yarp::sig::Vector& out, double& timestamp) const override;
+
+    /* IThreeAxisLinearAccelerometers methods */
+
+    size_t getNrOfThreeAxisLinearAccelerometers() const override;
+
+    yarp::dev::MAS_status getThreeAxisLinearAccelerometerStatus(size_t sens_index) const override;
+
+    bool getThreeAxisLinearAccelerometerName(size_t sens_index, std::string &name) const override;
+
+    bool getThreeAxisLinearAccelerometerFrameName(size_t sens_index, std::string &frameName) const override;
+
+    bool getThreeAxisLinearAccelerometerMeasure(size_t sens_index, yarp::sig::Vector& out, double& timestamp) const override;
+
+    /* IThreeAxisMagnetometers methods */
+
+    size_t getNrOfThreeAxisMagnetometers() const override;
+
+    yarp::dev::MAS_status getThreeAxisMagnetometerStatus(size_t sens_index) const override;
+
+    bool getThreeAxisMagnetometerName(size_t sens_index, std::string &name) const override;
+
+    bool getThreeAxisMagnetometerFrameName(size_t sens_index, std::string &frameName) const override;
+
+    bool getThreeAxisMagnetometerMeasure(size_t sens_index, yarp::sig::Vector& out, double& timestamp) const override;
+
+    /* IOrientationSensors methods */
+
+    size_t getNrOfOrientationSensors() const override;
+
+    yarp::dev::MAS_status getOrientationSensorStatus(size_t sens_index) const override;
+
+    bool getOrientationSensorName(size_t sens_index, std::string &name) const override;
+
+    bool getOrientationSensorFrameName(size_t sens_index, std::string &frameName) const override;
+
+    bool getOrientationSensorMeasureAsRollPitchYaw(size_t sens_index, yarp::sig::Vector& rpy, double& timestamp) const override;
 
 
 private:
+
+    yarp::dev::MAS_status genericGetStatus(size_t sens_index) const;
+    bool genericGetSensorName(size_t sens_index, std::string &name) const;
+    bool genericGetFrameName(size_t sens_index, std::string &frameName) const;
+    bool genericGetMeasure(size_t sens_index, yarp::sig::Vector& out, double& timestamp, size_t startIdx) const;
+
     yarp::sig::Vector m_imuData; //buffer for imu data
     yarp::os::Stamp m_lastTimestamp; //buffer for last timestamp data
-    yarp::os::Semaphore m_dataMutex; //mutex for accessing the data
+    mutable yarp::os::Semaphore m_dataMutex; //mutex for accessing the data
+    std::string m_sensorName{"sensor_imu_gazebo"};
```

Comment:
Gazebo has a Sensor::Name method, I think we can use that one for both the sensor and the frame, at least for now. We can read it in the `Load` and then pass it to the device via the yarp parameters.

Reply:
Done :+1:

Reply:
Sorry, I think it makes more sense (for eventual consistency with the real robot) to use the `::Name` method, not the `ScopedName` that will contains also the model name and the world name.

Reply:
I used the `_sensor->Name()` check line 145 of IMU.cc

Reply:
You are right, I was confused by the old code that used `ScopedName` instead, thanks!